src/NSL/LinAlg/minMax.tpp

```diff
@@ -0,0 +1,18 @@
+#ifndef NSL_LINALG_MINMAX_TPP
+#define NSL_LINALG_MINMAX_TPP
+
+namespace NSL::LinAlg{
+
+template<NSL::Concept::isNumber Type>
+NSL::Tensor<Type> min(const NSL::Tensor<Type> & t){
+    return torch::min(t);
+}
+
+template<NSL::Concept::isNumber Type>
+NSL::Tensor<Type> max(const NSL::Tensor<Type> & t){
+    return torch::max(t);
+}
```

Comment:
What happens if these are called on a singleton tensor?  Just return the contents?

Reply:
I didn't bother to overload a Type based min,max but we could add it here. So far we never needed it though.

Reply:
The return types of these should be `Type` and not `NSL::Tensor<Type>`, because they are reductions.  We could later add overloads that act along particular directions.

We should remember for the future that complex Types can compile correctly but will throw a runtime error.

Reply:
Done with commit 3accef4702951fa2dba3d8546378444e6a572aa2